proxy/src/connection.rs

```diff
@@ -32,19 +33,49 @@ pub struct Connecting(TcpStreamNew);
 /// socket to reduce the chance of TLS protections being accidentally
 /// subverted.
 #[derive(Debug)]
-pub enum Connection {
+pub struct Connection {
+    io: Io,
+    /// This buffer gets filled up when "peeking" bytes on this Connection.
+    ///
+    /// This is used instead of MSG_PEEK in order to support TLS streams,
+    /// which will need to decrypt the bytes before being able to "peek".
```

Comment:
The issue isn't exactly that the TLS stream will need to decrypt the bytes before being able to "peek", as far as I understand it. I suggest we just delete the second half of the sentence, rather than get into the specifics.